### DISCUSSION TITLE
What is the correct way to share a virtual environment among multiple projects

### DISCUSSION THREAD

**Discussion by idnsunset** [pdm-project/pdm#2612] [Q&A] [answered]
I have several projects that use the same virtual environment. According to the docs, it seems that I could run `pdm venv --name` to create a virtual environment at a specific location given by `venv.location`, but this way the name of the created virtual environment would be `<project_name>-<path_hash>-<name_or_python_version>` where the `<project_name>` gives an implication that this environment is specific to `<project_name>` but not shared by several projects, which is not that perfect.

I want to make sure if I could use `pdm` to manage such "out-of-project" virtual environments, or must I create a virtual environment  separately using other tools and then run `pdm use -f` to use it?

**Comment by frostming** [comment]
Create venv outside and run `pdm use /path/to/venv/bin/python`

> or must I create a virtual environment separately using other tools and then run `pdm use -f` to use it?


So yes

**Reply by idnsunset** [reply, marked as answer]
@frostming, thanks for your quick response!
